code/game/turfs/simulated/floor/plating.dm

```diff
@@ -0,0 +1,241 @@
+					ChangeTurf(/turf/simulated/floor/engine)
+					playsound(src, 'sound/items/Deconstruct.ogg', 80, 1)
+					R.use(2)
+					user << "<span class='notice'>You reinforce the floor.</span>"
+				return
+
+	else if(istype(C, /obj/item/stack/tile))
+		if(!broken && !burnt)
+			var/obj/item/stack/tile/W = C
+			if(!W.use(1))
+				return
+			ChangeTurf(W.turf_type)
+			playsound(src, 'sound/weapons/Genhit.ogg', 50, 1)
+		else
+			user << "<span class='warning'>This section is too damaged to support a tile! Use a welder to fix the damage.</span>"
+
+	else if(istype(C, /obj/item/weapon/weldingtool))
+		var/obj/item/weapon/weldingtool/welder = C
+		if( welder.isOn() && (broken || burnt) )
+			if(welder.remove_fuel(0,user))
+				user << "<span class='danger'>You fix some dents on the broken plating.</span>"
+				playsound(src, 'sound/items/Welder.ogg', 80, 1)
+				icon_state = icon_plating
+				burnt = 0
+				broken = 0
+
+/turf/simulated/floor/plating/airless
+	icon_state = "plating"
+	name = "airless plating"
+	oxygen = 0.01
+	nitrogen = 0.01
+	temperature = TCMB
+
+/turf/simulated/floor/plating/airless/New()
+	..()
+	name = "plating"
+
+/turf/simulated/floor/plating/airless/catwalk
+	icon = 'icons/turf/catwalks.dmi'
+	icon_state = "Floor3"
+	name = "catwalk"
+	desc = "Cats really don't like these things."
+
+/turf/simulated/floor/engine
+	name = "reinforced floor"
+	icon_state = "engine"
+	thermal_conductivity = 0.025
+	heat_capacity = 325000
+	floor_tile = /obj/item/stack/rods
+
+/turf/simulated/floor/engine/break_tile()
+	return //unbreakable
+
+/turf/simulated/floor/engine/burn_tile()
+	return //unburnable
+
+/turf/simulated/floor/engine/make_plating(var/force = 0)
+	if(force)
+		..()
+	return //unplateable
+
+/turf/simulated/floor/engine/attackby(obj/item/weapon/C as obj, mob/user as mob, params)
+	if(!C || !user)
+		return
+	if(istype(C, /obj/item/weapon/wrench))
+		user << "<span class='notice'>You begin removing rods...</span>"
+		playsound(src, 'sound/items/Ratchet.ogg', 80, 1)
+		if(do_after(user, 30, target = src))
+			if(!istype(src, /turf/simulated/floor/engine))
+				return
+			new /obj/item/stack/rods(src, 2)
+			ChangeTurf(/turf/simulated/floor/plating)
+			return
+
+/turf/simulated/floor/engine/ex_act(severity,target)
+	switch(severity)
+		if(1)
+			if(prob(80))
+				ReplaceWithLattice()
+			else if(prob(50))
+				qdel(src)
+			else
+				make_plating(1)
+		if(2)
+			if(prob(50))
+				make_plating(1)
+
+/turf/simulated/floor/engine/cult
+	name = "engraved floor"
+	icon_state = "cult"
+
+/turf/simulated/floor/engine/cult/narsie_act()
+	return
+
+/turf/simulated/floor/engine/n20/New()
+	..()
+	var/datum/gas_mixture/adding = new
+	var/datum/gas/sleeping_agent/trace_gas = new
+
+	trace_gas.moles = 6000
+	adding.trace_gases += trace_gas
+	adding.temperature = T20C
+
+	assume_air(adding)
+
+/turf/simulated/floor/engine/singularity_pull(S, current_size)
+	if(current_size >= STAGE_FIVE)
+		if(prob(30))
```

Comment:
well the engine tile actually will be unaffected because it calls `make_plating` with no arguments